src/components/pages/ImageClassification/ImageAnnotationModal/ImageAnnotationModalMap.js

```diff
@@ -435,7 +443,29 @@ const ImageAnnotationModalMap = ({
 
     const showFeaturePopupOnClick = ({ features }) => {
       const feature = features[0]
-      selectFeature(feature)
+      const { popupAnchorLngLat } = selectFeature(feature)
+
+      const bufferedFeature = buffer(feature, 1500) // the buffer unit is set by feel for laptop sized screens, it may need to be tweaked as we go
+      const mapBoundsFeature = bboxPolygon(map.current.getBounds().toArray().flat())
+      const isBufferedFeatureCompletelyWithinMapBounds = booleanContains(
+        mapBoundsFeature,
+        bufferedFeature,
+      )
+      const shouldAlsoZoom = map.current.getZoom() > 4.5
+      const easeToOptions = {
+        center: popupAnchorLngLat,
+        duration: DEFAULT_MAP_ANIMATION_DURATION,
+      }
+      if (shouldAlsoZoom) {
+        // Setting null zoom on easeTo will zoom all the way out;
+        // setting undefined will cause errors, so we need to make
+        // the zoom setting conditional.
+        easeToOptions.zoom = 4.5
```

Comment:
Things like this are what fuel my comments about mapbox being kinda junky software. They dont handle null or undefined values well at all. If there are error messages they are super weird.